add Number of old messages to sidebar_format

jindraj opened the pull request:
* **What does this PR do?**
Adds number of old messages as `%o` to `sidebar_format` variable.
Counting behaves the same like for `%o` in `status_format`

* **Screenshots (if relevant)**
[![asciicast](https://asciinema.org/a/WiOjaa5g9QumpUrGfCjrtyTqu.svg)](https://asciinema.org/a/WiOjaa5g9QumpUrGfCjrtyTqu)

* **Example**
Initial state: there are 8 unread messages in a mailbox: 5 new (unread, not seen), 3 old (unread, already seen)

| first time visit | %N | %o |revisiting mailbox | %N | %o |
| ---                    | ---  | --- |---                    | ---  | --- |
| mark_old=yes |     8 |    3 | mark_old=yes |     8 |    8 | 
| mark_old=no  |     8 | N/A |mark_old=no  |     8 | N/A |

Austin-Ray commented on sidebar.c:
```diff
@@ -195,6 +196,16 @@ static const char *sidebar_format_str(char *buf, size_t buflen, size_t col, int
+      if (!optional)
+      {
+        snprintf(fmt, sizeof(fmt), "%%%sd", prec);
+        snprintf(buf, buflen, fmt, m->msg_unread - m->msg_new );
```
You'll probably want to use `msg_count` instead of `msg_unread` since `msg_new` does not include old, unread emails.

jindraj replied:
I understand the concept of old messages and as mark_old works:
`set mark_old=no`
NEW = UNREAD

`set mark_old = yes`
NEW = unseen UNREAD 
OLD = already seen UNREAD

so `unread – new` seems right to me.

jindraj replied:
Sorry. I wrote the explanation wrong. NEW = unseen UNREAD + already seen UNREAD.

the behaviour is consistent with `status_format ` expando `%o`. The code is almost the same as  mentioned status_format https://github.com/neomutt/neomutt/blob/master/status.c#L240

jindraj replied:
… or simly NEW = UNREAD no matter what is `mark_old` set to.

jindraj commented:
> What's the use-case for adding this new expando?

It's similar to `%o` in `status_format`. There are people using neomutt with `mark_old` option turned on. It makes sense they want to distinguish between NEW new messages and OLD new messages.

I'm not using it personally, but there was a someone asking on IRC about this.

flatcap commented on doc/manual.xml.head:
```diff
@@ -792,6 +792,13 @@ set sidebar_divider_char = '│'          <emphasis role="comment"># Pretty line
+                    <entry>%o</entry>
+                    <entry>* †</entry>
+                    <entry>
+                      Number of New messages in the mailbox
```
er, 'New'?

flatcap commented:
> Initial state: there are 8 unread messages in a mailbox:
> 5 new (unread, not seen), 3 old (unread, already seen)

Thanks for the example.
It strengthens my idea that we're missing an expando for **New only**.

If I have a Mailbox with 3 old (unread, already seen) emails, then a new email arrives:
- `%N` displays 4
- `%o` displays 3
- Sidebar show the **new** colour

The behaviour of `%N` means that it can't be combined with `%o` (because it implies that there are 7 emails).
If we had a `%Z` (really new, not including seen emails), then we'd have:
- `%Z` displays 1
- `%o` displays 3
- Sidebar shows the **new** colour

That means I get the best of both worlds.  I can see exactly how many emails have arrived since I last looked and how many others that I haven't read.

**Expandos**:
- `%Z` New (unread, unseen)
- `%o` Old (unread, seen)
- `%u` Unread (seen or unseen)
- `%r` Read
- `%S` Total emails

**Mailbox has**:
- 5 unread emails
- 2 read emails

| Action          | `%Z` New | `%o` Old | `%u` Unread | `%r` Read | `%S` Total | Sidebar colour |
| :-------------- | :------- | :------- | :---------- | :-------- | :--------- | :------------- |
| Initial state   | 0        | 5        | 5           | 2         | 7          | UNREAD         |
| 3 new emails    | 3        | 5        | 8           | 2         | 10         | NEW            |
| Visit Mailbox   | 0        | 8        | 8           | 2         | 10         | UNREAD         |
| Read all emails | 0        | 0        | 0           | 10        | 10         | ORDINARY       |

This gives the user some flexibility, e.g.

| Action          | Unread/Total<BR>`%u/%S` | New/Old<BR>`%Z/%o` | New/Old/Read<BR>`%Z/%o/%r` | Sidebar colour |
| :-------------- | :---------------------- | :----------------- | :------------------------- | :------------- |
| Initial state   | 5/7                     | 0/5                | 0/5/2                      | UNREAD         |
| 3 new emails    | 8/10                    | 3/5                | 3/5/2                      | NEW            |
| Visit Mailbox   | 8/10                    | 0/8                | 0/8/2                      | UNREAD         |
| Read all emails | 0/10                    | 0/0                | 0/0/10                     | ORDINARY       |

(examples are for the case: `mark_old=yes`)

jindraj commented:
added
 - **%Z** expando to  both sidebar_format  and status_format which shows _msg_new_
OK, It seems there's no need for **%Z** in _status_format_ since **%n** does the same.

 - **%r** expando to sidebar_format *msg_count - msg_unread*

I removed the malibox name from status format for better overview. 
[![asciicast](https://asciinema.org/a/YAmdZpwyErll52YLHvyTMihl6.svg)](https://asciinema.org/a/YAmdZpwyErll52YLHvyTMihl6)
I recreated the state of your table 5 OLD (UNREAD, SEEN) messages. With the only difference. There's 20 msgs total.
We're inside the first mailbox of the sidebar all the time. I just reopen the mailbox in the time 13s to show the state after reopening the mailbox and transition of the counts of NEW (unseen) to OLD (new seen)

config used:
```
set mark_old
set sidebar_format="Z:%Z o:%o N:%N r:%r %*  S:%5S "
set status_format = "%D Z:%Z n:%n o:%o u:%u R:%R m:%m " # I cut the rest since it's not important
```

flatcap commented:
**Lookup Table**

| Description             | Sidebar | Status | Mailbox Calculation      |
| :---------------------- | :------ | :----- | :----------------------- |
| New (unread, unseen)    | `%Z`    | `%n`   | `msg_new`                |
| Old (unread, seen)      | `%o`    | `%o`   | `msg_unread - msg_new`   |
| Unread (seen or unseen) | `%N`    | `%u`   | `msg_unread`             |
| Read                    | `%r`    | `%R`   | `msg_count - msg_unread` |
| Total emails            | `%S`    | `%m`   | `msg_count`              |

flatcap commented on status.c:
```diff
@@ -376,6 +376,20 @@ static const char *status_format_str(char *buf, size_t buflen, size_t col, int c
+               (Context->mailbox->msg_new == 0))
+        optional = false;
+      break;
+
```
It seems that we don't need `%Z` since `%n` already does the same

flatcap commented:
I've rebased, squashed and tidied (a little whitespace) your commits.
I've also added a commit that brings the rest of the sidebar docs up-to-date.
(there are no functional changes to your code)
